OneSignalSDK/onesignal/core/src/main/java/com/onesignal/user/IUserStateObserver.kt

```diff
@@ -0,0 +1,15 @@
+package com.onesignal.user
+
+/**
+ * A user state changed handler. Implement this interface and provide the implementation
+ * to be notified when the user has changed.
+ */
+interface IUserStateObserver {
```

Comment:
We should call out somewhere in the KDoc that app developers should be checking **both** the `onesignalId` **and** the `externalId` in the `UserState` to make sure they are grabbing the correct `onesignalId`.

For example, a new app install that calls `login("ext1")` will trigger the callback twice. App developers may just grab the very first callback and save the `onesignalId`. However, that first callback will contain an `""` `externalId`, and it is the second callback that contains the actual `onesignalId` they want and an `externalId` of `ext1`.

I'm not sure where best to put this information since it can go in the description of `IUserStateObserver` or `fun onUserStateChange`, `UserChangedState`, or `UserState`. 

I need to do the equivalent for iOS as well.